src/main/resources/db/migration/V0002__purge.sql

```diff
@@ -0,0 +1,209 @@
+DECLARE i_purge_job_id NUMERIC;
+
+  -- cursor1 service requests created before given retention date
+  c1 CURSOR (n_param DOUBLE PRECISION)
+            FOR SELECT service_request_id
+               FROM sdt_owner.service_requests
+               WHERE request_timestamp < t_retention_date;
+
+  -- cursor2 bulk submissions with service requests created before given retention date
+  c2 CURSOR (n_param DOUBLE PRECISION)
+            FOR SELECT bulk_submission_id
+               FROM sdt_owner.bulk_submissions
+               WHERE service_request_id IN (
+                   SELECT  service_request_id
+                   FROM sdt_owner.service_requests
+                   WHERE request_timestamp < t_retention_date);
+
+  -- cursor3 individual requests for bulk submissions
+  --   with service requests created before given retention date
+  c3 CURSOR (n_param DOUBLE PRECISION)
+            FOR SELECT individual_request_id,sdt_request_reference
+               FROM sdt_owner.individual_requests
+               WHERE bulk_submission_id IN  (
+                   SELECT bulk_submission_id
+                   FROM sdt_owner.bulk_submissions
+                   WHERE service_request_id IN (
+                       SELECT service_request_id
+                       FROM sdt_owner.service_requests
+                       WHERE request_timestamp < t_retention_date));
+
+  -- cursor4 error logs for individual requests for bulk submissions
+  --   with service requests created before given retention date
+  c4 CURSOR (n_param DOUBLE PRECISION)
+            FOR SELECT error_log_id
+               FROM sdt_owner.error_logs
+               WHERE individual_request_id IN (
+                   SELECT individual_request_id
+                   FROM sdt_owner.individual_requests
+                   WHERE bulk_submission_id IN (
+                       SELECT bulk_submission_id
+                       FROM sdt_owner.bulk_submissions
+                       WHERE service_request_id IN (
+                           SELECT service_request_id
+                           FROM sdt_owner.service_requests
+                           WHERE request_timestamp < t_retention_date)));
+
+  n_iteration   INT;
+  BEGIN
+
+    -- retention period
+	SELECT parameter_value
+    INTO i_retention_period
+    FROM sdt_owner.global_parameters
+    WHERE parameter_name = 'DATA_RETENTION_PERIOD';
+
+    -- job start date
+	SELECT now()
+    INTO t_job_start_date;
+    RAISE NOTICE 'Job start date: %', t_job_start_date;
+
+    -- retention_date calculated using retention period and now
+	SELECT t_job_start_date::DATE - (i_retention_period || ' DAY')::INTERVAL
+    INTO t_retention_date;
+    RAISE NOTICE 'Retention period date: %', t_retention_date;
+
+    -- create job audit
+	INSERT INTO sdt_owner.purge_job_audit
+  	( job_start_date, success)
+  	VALUES (t_job_start_date, 0);
+
+    -- get job audit id
+	SELECT max(purge_job_id)
+    INTO i_purge_job_id
+    FROM sdt_owner.purge_job_audit;
```

Comment:
Will there be a performance impact in using this method to get the current job id when the table has a lot of data in it?  If so might be better to use a sequence for setting the purge job id.

Reply:
Done